### PR TITLE
Fetch cluster info from Metadata server

### PR DESCRIPTION
**Pull request by elfinhe**
Read GKE_CLUSTER_URL from GCP Metadata server

[ ] Configuration Infrastructure
[ ] Docs
[x] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[x] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Comment by costinm** [comment]
https://github.com/istio/istio/pull/26482 has a similar change - but in tokenmanager.

If I manage to merge it - could you use that method ? I have another PR doing the other half - but happy to use your PR instead.

**Comment by elfinhe** [comment]
> #26482 has a similar change - but in tokenmanager.
> 
> If I manage to merge it - could you use that method ? I have another PR doing the other half - but happy to use your PR instead.

Let's use the one in platform/gcp.go in this PR. looks like it is not specific to tokenmanager. yes, both of them have the similar logics. So shouldn't be hard to merge.

**Comment by elfinhe** [comment]
> > #26482 has a similar change - but in tokenmanager.
> > If I manage to merge it - could you use that method ? I have another PR doing the other half - but happy to use your PR instead.
> 
> Let's use the one in platform/gcp.go in this PR. looks like it is not specific to tokenmanager. yes, both of them have the similar logics. So shouldn't be hard to merge.

Looks #26482 is merged, I am happy to merge them in this PR.

File: security/pkg/nodeagent/caclient/providers/google/client.go [hunk 1]
```diff
@@ -37,7 +38,7 @@ const bearerTokenPrefix = "Bearer "
 
 var (
 	googleCAClientLog = log.RegisterScope("googleca", "Google CA client debugging", 0)
-	gkeClusterURL     = env.RegisterStringVar("GKE_CLUSTER_URL", "", "The url of GKE cluster").Get()
+	envGkeClusterURL  = env.RegisterStringVar("GKE_CLUSTER_URL", "", "The url of GKE cluster").Get()
```
Comment (JimmyCYJ) [inline security/pkg/nodeagent/caclient/providers/google/client.go:41]: envGKEClusterURL

**Pull request merged**